### PR TITLE
RAD-87 Update guide star information, catalog version, and science fi…

### LINKED ISSUE #177: Add/Update guide star attribute information
_Issue [RAD-87](https://jira.stsci.edu/browse/RAD-87) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

Need to add

gs_id: The guide star identifier is a 10 character, base 36 number identifying celestial objects included in Guide Star Catalog II, e.g. N5R2005564

gs_order: The index of this guide star within the list of selected guide stars. Note: SDF should set this from METRICS_DB:GsAcq.guideStarOrder row for visitId and SUCCESSFUL executionStatus

Guide Star catalog number / version
Note: this is an input to select the catalog from the web service, e.g. "GSC 2.4.2" and is also returned as a string in the query (GSC242)

Guide window file name for L1 files
This will be set by SDF and set the file name of the detector that this guide window is associated with,
e.g. r0000101001001001001_01101_0001_WFI01_uncal.asdf

### PR DESCRIPTION
**Pull request by ddavis-stsci**
…le reference

Resolves [RAD-87](https://jira.stsci.edu/browse/RAD-87)

Closes #177 

This PR adds the guide star catalog id, catalog version where the information is obtained, 
and science file name that is associated with the guide window information. 
The unit tests pass
```
pytest tests/
=================================================================== test session starts ===================================================================
platform darwin -- Python 3.11.0, pytest-7.3.1, pluggy-1.0.0
rootdir: /Users/ddavis/src/Roman/rad
configfile: pyproject.toml
plugins: remotedata-0.4.0, hypothesis-6.75.1, asdf-2.15.0, mock-3.10.0, filter-subpackage-0.1.2, astropy-header-0.2.2, doctestplus-0.12.1, astropy-0.10.0, cov-4.0.0, ci-watson-0.6.1, openfiles-0.5.0, arraydiff-0.5.0
collected 397 items

tests/test_integration.py ..........................................................                                                                [ 14%]
tests/test_manifest.py .                                                                                                                            [ 14%]
tests/test_schemas.py ............................................................................................................................. [ 46%]
................................................................................................................................................... [ 83%]
..................................................................                                                                                  [100%]

=================================================================== 397 passed in 2.63s
```

**Checklist**
- [ ] Schema changes discussed at RAD Review Board meeting
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [x] Updated relevant roman_datamodels utilities and tests
- [ ] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/XXX/

### CODE REVIEW

**Review by PaulHuwe — APPROVED** [review]
LGTM

File: src/rad/resources/schemas/guidestar-1.0.0.yaml [hunk 1]
```diff
@@ -25,6 +25,40 @@ properties:
...
+  gw_science_file_source:
+    title: The science data associated with this guide window
+    description: The science data file that is associated with this guide window, e.g. "r0000101001001001001_01101_0001_WFI01_uncal.asdf"
+    type: string
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: TBD
+    archive_catalog:
+      datatype: nvarchar(50)
```
[the remark this answers is not in the document]
Comment (ddavis-stsci) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:37]: I upped it from 50 to 120 to match the length in the filename schema.

File: src/rad/resources/schemas/guidestar-1.0.0.yaml [hunk 2]
```diff
@@ -25,6 +25,40 @@ properties:
...
+    title: guide star identifier from the GSC2 catalog
+    description: guide star catalog id from the GSC2, field gsc2ID
+    type: string
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: TBD
+    archive_catalog:
+      datatype: nvarchar(20)
+      destination: [ScienceCommon.gs_id]
```
[the remark this answers is not in the document]
Comment (ddavis-stsci) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:49]: Good questions and I'm not sure. I'll have to check with Chris H and see if that information is available in the telemetry for the guidewindow data. I think we can get it from pss in the dms_guide_star table if needed.

File: src/rad/resources/schemas/guidestar-1.0.0.yaml [hunk 3]
```diff
@@ -25,6 +25,40 @@ properties:
...
+    description: The version of the catalog that the guide stars are selected, currently  "GSC 2.4.2", SDF should populate
+        this from the return value of CAT e.g. CAT=GSC242
+    type: string
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: TBD
+    archive_catalog:
+      datatype: nvarchar(20)
+      destination: [ScienceCommon.gs_catalog_version]
```
[the remark this answers is not in the document]
Comment (ddavis-stsci) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:61]: I don't think that is needed in the guidewindow files.

File: src/rad/resources/schemas/guidestar-1.0.0.yaml [hunk 4]
```diff
@@ -25,6 +25,40 @@ properties:
...
+    title: The science data associated with this guide window
+    description: The science data file that is associated with this guide window, e.g. "r0000101001001001001_01101_0001_WFI01_uncal.asdf"
+    type: string
+    sdf:
+      special_processing: VALUE_REQUIRED
+      source:
+        origin: TBD
+    archive_catalog:
+      datatype: nvarchar(50)
+      destination: [ScienceCommon.gw_science_file_source]
```
[the remark this answers is not in the document]
Comment (ddavis-stsci) [reply src/rad/resources/schemas/guidestar-1.0.0.yaml:38]: good catch I'll move it.

**Pull request merged**